### PR TITLE
refactor spline so an anonymous function can be returned for use elsewhere

### PR DESCRIPTION
**Pull request by CamDavidsonPilon**
Previously my code might look like:
```python
rossi_with_splines[['age0', 'age1']] = spline(rossi_with_splines, var='age', term=2, restricted=True)
cph = CoxPHFitter().fit(rossi_with_splines.drop('age', axis=1), 'week', 'arrest')

# this part is nasty
df =rossi_with_splines.drop_duplicates('age').sort_values('age')[['age', 'age0', 'age1']].set_index('age')
(df * cph.hazards_[['age0', 'age1']].values).sum(1).plot()
```

vs 

```python
spline_transform, _ = create_spline_transform(df['age'], term=2, restricted=True)
rossi_with_splines[['age0', 'age1']] = spline_transform(rossi_with_splines['age'])

cph = CoxPHFitter().fit(rossi_with_splines.drop('age', axis=1), 'week', 'arrest')

ages_to_consider = np.arange(20, 45))
y = spline_transform(ages_to_consider).dot(cph.hazards_[['age0', 'age1']].values)
plot(ages_to_consider, y)
```

### CODE REVIEW

**Comment by pzivich** [comment]
Hmmm for some reason travis is not finding `numpy.quantile` for the splines. I will look into why it is failing.

Could you add a 'References' page to ReadTheDocs for the new `creat_spline_transform` function?

**Comment by pzivich** [comment]
Alright, it is an install issue. `np.quantile` was added in 1.15. TravisCI was installing NumPy version 1.13.1 for the Python 3.6 test

**Comment by CamDavidsonPilon** [comment]
Hm, it's strange that only a one CI run was using a different numpy, no? 

Also, I think numpy >= 1.15 is too strict. We should think of another solution to that (maybe pandas, but I want to avoid it if I can)

**Comment by pzivich** [comment]
The solution looks like numpy `percentile()`. It goes back to at least v1.9 of numpy.

Those two instances are the only two calls in zEpid. They could be replaced with `np.percentile(array, q=knots)`. However the `q` argument would need to be modified

File: zepid/base.py [hunk 1]
```diff
@@ -1676,6 +1676,82 @@ def interaction_contrast_ratio(df, exposure, outcome, modifier, adjust=None, reg
...
+        providing an unrestricted spline
+
+    Returns
+    ---------
+    function : a lambda function that accepts a numpy array or
+    list : a list of the knots
+    """
+    if knots is None:
+        if n_knots == 1:
+            knots = [0.5]
```
Comment (pzivich) [inline zepid/base.py]: If switching to `np.percentile`, this would become `knots = [50]`. Same for the others

Comment (pzivich) [reply zepid/base.py]: This wouldn't change how the `knots` optional argument works. That argument takes a list of the actual knot placements, so it should be all good

**Comment by pzivich** [comment]
I will pull this, update to `percentiles`, and push it to master today/tomorrow. It will be part of the v0.6.0 release

**Pull request merged**